### PR TITLE
Fix React StrictMode useCollection

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 

Unsure if this would be correct, but I noticed we don't use the factory's `prev` argument anywhere. I have no idea what it would be used for, so I don't know if removing it is really ok. It does reduce the number of failed tests.

Note on tests, failure count may be inaccurate due to cascading failures
Before
Test Suites: 27 failed, 1 skipped, 144 passed, 171 of 172 total
Tests:       657 failed, 49 skipped, 3534 passed, 4240 total
Snapshots:   1 passed, 1 total

After
Test Suites: 27 failed, 1 skipped, 144 passed, 171 of 172 total
Tests:       657 failed, 49 skipped, 3534 passed, 4240 total
Snapshots:   1 passed, 1 total

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-stately/collections/src/useCollection.ts [hunk 1]
```diff
@@ -12,19 +12,16 @@
...
-    prev.current = factory(nodes, prev.current);
-    return prev.current;
-  // Don't invalidate when any prop changes, just the two we care about.
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [builder, props.children, props.items, context, ...invalidators]);
+  let {children, items} = props;
+  let collection = useMemo(() => {
+    let nodes = builder.build({children, items}, context);
+    return factory(nodes);
+  }, [builder, children, items, context, factory, ...invalidators]);
```
Comment (devongovett) [inline packages/@react-stately/collections/src/useCollection.ts:25]: Do we need invalidators anymore if we are using useCallback around factory with the same invalidators?

File: packages/@react-stately/collections/src/useCollection.ts [hunk 2]
```diff
@@ -12,19 +12,16 @@
 
 import {Collection, CollectionBase, Node} from '@react-types/shared';
 import {CollectionBuilder} from './CollectionBuilder';
-import {useMemo, useRef} from 'react';
+import {useMemo} from 'react';
 
-type CollectionFactory<T, C extends Collection<Node<T>>> = (node: Iterable<Node<T>>, prev: C | null) => C;
+type CollectionFactory<T, C extends Collection<Node<T>>> = (node: Iterable<Node<T>>) => C;
```
Comment (devongovett) [inline packages/@react-stately/collections/src/useCollection.ts:17]: Technically a breaking change?

**Review by devongovett — APPROVED** [review]

File: packages/@react-stately/collections/src/useCollection.ts (hunk 2 above)
Comment (snowystinger) [reply packages/@react-stately/collections/src/useCollection.ts:17]: only in so far as i removed the parameter... it wasn't being used, so don't think it's breaking. and passing something to it won't cause anything to error or break, it'll just be ignored, as it already was

File: packages/@react-stately/collections/src/useCollection.ts (hunk 1 above)
Comment (snowystinger) [reply packages/@react-stately/collections/src/useCollection.ts:25]: technically not i guess, but that is definitely breaking

**Review by devongovett — APPROVED** [review]

**Review by reidbarber — APPROVED** [review]

**Pull request merged**